### ISSUE TITLE
TrailingSemicolon with Sass Map

### ISSUE THREAD

**Issue by AESM** [brigade/scss-lint#608]
My `scss-lint` keeps giving me the following warning, apparently thinking there is no semicolon after the declaration's property-value:

`[W] TrailingSemicolon: Declaration should be terminated by a semicolon`

I have provided a few examples.

_Case 1A_ shows the warning, but works the same as _Case 1B_.

_Case 1A_:
``` scss
$white: #FFFFFF;
$grey: #808080;
$black: #0B0B0B;

$achromatic-palette: (  // [W] found here
  white-mod: (
    light: lighten($white, 8%),
    base: $white,
    dark: darken($white, 8%)
  ),
  grey-mod: (
    light: lighten($grey, 8%),
    base: $grey,
    dark: darken($grey, 8%)
  ),
  black-mod: (
    light: lighten($black, 8%),
    base: $black,
    dark: darken($black, 8%)
  )
);
```

_Case 1B_ passes, but is less human-readable than _Case 1A_ (in my opinion).

_Case 1B_:
``` scss
$white: #FFFFFF;
$grey: #808080;
$black: #0B0B0B;

// no [Warning] found
$achromatic-palette: (
  white-mod:
  (light: lighten($white, 8%),
    base: $white,
    dark: darken($white, 8%)),
  grey-mod:
    (light: lighten($grey, 8%),
    base: $grey,
    dark: darken($grey, 8%)),
  black-mod:
    (light: lighten($black, 8%),
    base: $black,
    dark: darken($black, 8%))
);
```

_Case 2A_ and _Case 2B_ are simplified examples to show the same results as _1A_ & _1B_.

_Case 2A_:
``` scss
$grey: #808080;

$achromatic-palette: (  // [W] found here
  grey-mod: (
    base: $grey
  )
);
```

_Case 2B_:
``` scss
$grey: #808080;

// no [W] found
$achromatic-palette: (
  grey-mod: (base: $grey)
);
```

_Case 3_ is an oversimplified example that does pass.  I'm including it as a test case.

_Case 3_:
``` scss
$grey: #808080;

// no [W] found
$achromatic-palette: (
  base: $grey
);
```

My **default.yml** file's `TrailingSemicolon` rule:
``` yaml
linters:
  TrailingSemicolon:
    enabled: true
```

My versions:
- `ruby 2.2.3p173`
- `Sass 3.4.18`
- `scss-lint 0.42.1`

**Comment by aju** [comment]
Passes:
```
$fontSizes: (
  header: 3rem
);
```

Not Passes
```
$fontSizes: (
  header: 3rem // 48px
);
```

Its an issue with comment in this case

**Comment by AESM** [comment]
This is not a comment issue.

@aju – Sorry if I didn't make it clear, but I only included the comments to point out where the warnings were showing up. No comments are actually contained within my work.

Also, the code you used is reminiscent of _Case 3_, the example I included to show that standard maps were indeed Passing, while long-form nested maps (_Case 1A_ & _Case 2A_) were receiving the `TrailingSemicolon` Warning.

**Comment by aju** [comment]
@AESM sorry I was not clear - my code show another examples that was working and now it broken :)

**Comment by ArmorDarks** [comment]
I confirm that comment in the end of map will throw error: `[W] TrailingSemicolon:  Declaration should be terminated by a semicolon`

Passes:

``` scss
$map: (
	test: ()
);
```

Not passes:

``` scss
$map: (
	test: () // comment
);
```

**Comment by AESM** [comment]
To lessen confusion, I want to point out that there are now two different `[W] TrailingSemicolon` issues on this thread: 

- The _long-form Sass map nesting_ issue I presented initially
- The _comments inside of a Sass map_ issue that @aju & @ArmorDarks presented

**Comment by ArmorDarks** [comment]
Sorry for hijacking your issue. I didn't realize at first place that you've described other issue.

I should create new issue then, right?

**Comment by richlocke** [comment]
I found this workaround worked for me:
```
$map: ( // Comment ending with a semi-colon;
	header: 3rem
);
```

* gem scss_lint version: `0.42.2`
* grunt package grunt-scss-lint version: `0.3.8`

**Comment by AESM** [comment]
@ArmorDarks – I'm only concerned that the two separate issues in a single ticket can get confusing because they are so similar, & that (because of all the referencing) my original issue may get overshadowed.

If the discussion can be kept clear & separate, I've got no problems.   : )

**Comment by Undistraction** [comment]
+1 The following is incorrectly picked up by the TrailingSemicolon linter:

```
$app-rhythm-value-map: (
  default: (
    vertical: 17px,
    horizontal: 17px
  )
);
```

**Comment by sds** [comment]
Thanks for the report, @AESM!

Looks like this was fixed by @srawlins in 8859b8a.

**Comment by AESM** [comment]
Much appreciated @sds & @srawlins!   : )
